### ISSUE TITLE
ConnectionError when deploying on Heroku with "Premium 0" Redis instance

### ISSUE THREAD

**Issue by toobaz** [oTree-org/otree-core#546]
I have been succesfully running an oTree install on Heroku for weeks, then (few hours before going into production 😏) I followed the advice to upgrade from the free Redis db to the "Premium 0" one. I immediately started getting errors when launching a session - more precisely, when the last participant enters the code for a room in a newly created session.

The error page I get is 

```
ConnectionError at /p/jrtid3pb/rpg/WelcomeWait/1/
Error 1 connecting to ec2-54-166-56-64.compute-1.amazonaws.com:20069. [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1076).
Request Method: GET
Request URL: https://scoutpg.herokuapp.com/p/jrtid3pb/rpg/WelcomeWait/1/
Django Version: 2.2.12
Exception Type: ConnectionError
Exception Value:
Error 1 connecting to ec2-54-166-56-64.compute-1.amazonaws.com:20069.
[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed
certificate in certificate chain (_ssl.c:1076).
Exception Location: /app/.heroku/python/lib/python3.7/site-packages/redis
/connection.py in connect, line 543
Python Executable: /app/.heroku/python/bin/python
Python Version: 3.7.7
Python Path:
['.',
'/app/.heroku/python/bin',
'/app',
'/app/.heroku/python/lib/python37.zip',
'/app/.heroku/python/lib/python3.7',
'/app/.heroku/python/lib/python3.7/lib-dynload',
'/app/.heroku/python/lib/python3.7/site-packages']
Server time: Ven, 27 Nov 2020 16:11:55 +0100
```

Now, my best guess is that this is a Heroku problem, because it started appearing without me changing a line of code, and because it appears randomly - sometimes the page loads fine.

The reason I signal this here is that the Heroku docs state ``"Hobby plans cannot be created using Redis versions earlier than 6, however production plans can be created using older supported versions."``... so there is some _intended_ difference between the different databases, performance apart, the Redis way to build secure connections did change from version 5 to 6, and I couldn't exclude that the HUEY queuing rules, rather than a Heroku sort of round robin, could be the source of indeterminacy.

... but please just close this if you think it is really Heroku's fault.

**Comment by oTree-org** [comment]
Thank you for reporting and investigating! in the past week or two, many people have started to report this issue to me (there is discussion on the mailing list: https://groups.google.com/g/otree). I really don't know what is causing it but maybe it's because oTree uses an older version of Huey and something recently changed on Heroku. So, I upgraded oTree's requirement from `huey==2.1.1` to `huey>=2.1.1`. Maybe put `huey==2.3.0` in your reqs and see if that fixes the problem?

**Comment by oTree-org** [comment]
FYI a user sent me this: 

"At a point where I deployed a premium Redis on Heroku, oTree would start to occasionally fail over on connection to redis. It seems that recently Heroku Redis started to authenticate their traffic and they use self-signed certificates. The error I have been getting on connection to redis occasionally was that self-signed certificates could not be verified. I moved to oTree Hub hoping that it may be a configuration issue, that I have forgotten to turn on some flag, but the issue seems to persist. Also, the issue seems to not be present when using free Heroku Redis."

I expect this error not to occur on oTree 3.3 or above.
